gobblin-modules/gobblin-compliance/src/main/java/gobblin/compliance/restore/RestorableHivePartitionDataset.java

```diff
@@ -0,0 +1,183 @@
+      HadoopUtils.setPermissions(getLocation().getParent(), this.datasetOwner, this.trashOwner, this.datasetOwnerFs,
+          permission);
+      log.info("Moved data from backup " + this.datasetToRestore.getLocation() + " to location " + getLocation());
+      executeDropPartitionQueries(queryExecutor);
+    }
+  }
+
+  private void executeTrashTableQueries(HiveProxyQueryExecutor queryExecutor)
+      throws IOException {
+    String trashTableName = getTrashTableName();
+    Preconditions.checkArgument(this.state.contains(ComplianceConfigurationKeys.TRASH_DB),
+        "Missing required property " + ComplianceConfigurationKeys.TRASH_DB);
+    String trashDbName = this.state.getProp(ComplianceConfigurationKeys.TRASH_DB);
+    try {
+      queryExecutor.executeQuery(HivePurgerQueryTemplate.getUseDbQuery(trashDbName), this.trashOwner);
+      queryExecutor.executeQuery(HivePurgerQueryTemplate
+          .getCreateTableQuery(trashDbName + "." + trashTableName, getDbName(), getTableName(),
+              getTrashTableLocation()), this.trashOwner);
+      queryExecutor.executeQuery(HivePurgerQueryTemplate
+          .getAddPartitionQuery(trashTableName, PartitionUtils.getPartitionSpecString(getSpec())), this.trashOwner);
+      queryExecutor.executeQuery(HivePurgerQueryTemplate
+          .getAlterTableLocationQuery(trashTableName, PartitionUtils.getPartitionSpecString(getSpec()),
+              getTrashPartitionLocation().toString()), this.trashOwner);
+    } catch (SQLException e) {
+      throw new IOException(e);
+    }
+  }
+
+  private void executeDropPartitionQueries(HiveProxyQueryExecutor queryExecutor)
+      throws IOException {
+    String dbName = this.datasetToRestore.getDbName();
+    String tableName = this.datasetToRestore.getTableName();
+    String partitionSpec = PartitionUtils.getPartitionSpecString(this.datasetToRestore.getSpec());
+    try {
+      queryExecutor.executeQuery(HivePurgerQueryTemplate.getUseDbQuery(dbName), this.datasetToRestoreOwner);
+      queryExecutor.executeQuery(HivePurgerQueryTemplate.getDropPartitionQuery(tableName, partitionSpec),
+          this.datasetToRestoreOwner);
+    } catch (SQLException e) {
+      throw new IOException(e);
+    }
+  }
+
+  private String getTrashTableName() {
+    return getCompleteTableName() + ComplianceConfigurationKeys.TRASH + this.timeStamp;
+  }
+
+  private void setTimeStamp() {
+    this.timeStamp = Long.toString(System.currentTimeMillis());
+  }
+
+  private String getCompleteTableName() {
+    return StringUtils.join(Arrays.asList(getDbName(), getTableName()), ComplianceConfigurationKeys.DBNAME_SEPARATOR);
+  }
+
+  private String getTrashTableLocation() {
+    Preconditions.checkArgument(this.state.contains(ComplianceConfigurationKeys.TRASH_DIR),
+        "Missing required property " + ComplianceConfigurationKeys.TRASH_DIR);
+    return this.state.getProp(ComplianceConfigurationKeys.TRASH_DIR) + getCompleteTableName();
```

Comment:
Is it better to use `new Path(this.state.getProp(ComplianceConfigurationKeys.TRASH_DIR), getCompleteTableName())`?